### DISCUSSION TITLE
Should I add a not-found.jsx file for each page?

### DISCUSSION THREAD

**Discussion by gundogduyakici** [vercel/next.js#47744] [Help]
### Summary

Hello,

I have a not-found.jsx file under the App folder, and on some pages I redirect to the notFound() function for some situations and I see the design. However, no link works. It does not redirect.

But if I add not-found.jsx in this page folder, it works fine. Do I have to add not-found.jsx to each page individually?

My Next.js version is 13.2.3.

**Comment by haydenbleasel** [comment]
@gundogduyakici With the release of Next.js 13.3...

> the root app/not-found.js file will also handle any unmatched URLs for your whole application. This means users that visit a URL that is not handled by your app will be shown the UI exported by the app/not-found.js file.

[Rearn more here](https://nextjs.org/blog/next-13-3).
